test/unit/remote-config/remote-config-api-client.spec.ts

```diff
@@ -415,26 +362,82 @@ describe('RemoteConfigApiClient', () => {
       stubs.push(stub);
       const expected = new FirebaseRemoteConfigError(
         'unknown-error', 'Unexpected response with status: 404 and body: not json');
-      return apiClient.publishTemplate(REMOTE_CONFIG_TEMPLATE)
+      return rcOperation()
         .should.eventually.be.rejected.and.deep.equal(expected);
     });
 
-    VALIDATION_ERROR_MESSAGES.forEach((message) => {
-      it('should reject with failed-precondition when a validation error occurres', () => {
-        const stub = sinon
-          .stub(HttpClient.prototype, 'send')
-          .rejects(utils.errorFrom({
-            error: {
-              code: 400,
-              message: message,
-              status: "FAILED_PRECONDITION"
-            }
-          }, 400));
-        stubs.push(stub);
-        const expected = new FirebaseRemoteConfigError('failed-precondition', message);
-        return apiClient.publishTemplate(REMOTE_CONFIG_TEMPLATE)
-          .should.eventually.be.rejected.and.deep.equal(expected);
+    it('should reject when rejected with a FirebaseAppError', () => {
+      const expected = new FirebaseAppError('network-error', 'socket hang up');
+      const stub = sinon
+        .stub(HttpClient.prototype, 'send')
+        .rejects(expected);
+      stubs.push(stub);
+      return rcOperation()
+        .should.eventually.be.rejected.and.deep.equal(expected);
+    });
+  }
+
+  function runTemplateVersionNumberTests(rcOperation: Function): void {
```

Comment:
Don't see why this needs to be a function. If you're planning for future reuse, I'd rather refactor this into a function at that point.

Reply:
Yeah, I am planing to use in the next PR. I will introduce the function at that point then :)